dali/pipeline/operator/op_schema.h

```diff
@@ -388,6 +411,75 @@ class DLL_PUBLIC OpSchema {
 
   DLL_PUBLIC string Dox() const;
 
+  DLL_PUBLIC bool CanUseDefaultInputDox() {
+    // TODO(klecki): add a flag to disallow this
+    if (MinNumInput() == 1 && MaxNumInput() == 1 && input_docs_.size() == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * @brief Check if this operator has input documentation specified
+   *
+   * If this is true `GetCallSignatureInputs` and `InputDox` can be used to obtain it
+   *
+   * @return DLL_PUBLIC HasInputDox
+   */
+  DLL_PUBLIC bool HasInputDox() {
+    if (static_cast<int>(input_docs_.size()) == MaxNumInput()) {
+      return true;
+    }
+    if (CanUseDefaultInputDox()) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * @brief List all the inputs that should appear in `__call__` signature based on the input
+   *        docs that were specified. Requires HasInputDox() to return true
+   *
+   * @return DLL_PUBLIC CallSignature
+   */
+  DLL_PUBLIC std::string GetCallSignatureInputs() {
+    DALI_ENFORCE(HasInputDox() && !CanUseDefaultInputDox(),
+                 "Input documentation was not specified for this operator.");
+    std::stringstream result;
+    for (int i = 0; i < MinNumInput(); i++) {
+      result << input_docs_[i].name;
+      if (i < MaxNumInput() - 1) {
+        result << ", ";
+      }
+    }
+    for (int i = MinNumInput(); i < MaxNumInput(); i++) {
```

Comment:
Doesn't `HasInputDox` require that all `MaxNumInput` are documented?

Reply:
It does, but I put `someting = None` to indicate that they are optional. (Didn't test this one yet).